seastar/src/timer.rs

```diff
@@ -305,6 +333,168 @@ impl LowresClock {
     }
 }
 
+/// Specialization of Seastar's timer that uses [`LowresClock`] as a clock.
```

Comment:
We have lots of code duplicated for all three types of clocks. I strongly suggest looking into declarative macros in order to generate slightly different implementations for all clock types.

I see this pattern sometimes when looking at the stdlib code. An example:

Macro definition: https://doc.rust-lang.org/src/core/num/int_macros.rs.html#1
Use: https://doc.rust-lang.org/src/core/num/mod.rs.html#241

Reply:
I made 2 macros:
- `timer_impl` that generates implementation of the timer specializations,
- `test_timer` that generates tests for the timer specializations.

The whole ffi part of the timer module still doesn't use macros. I don't think that using macros inside the `#[cxx::bridge]` module is possible. I didn't use macros in `timer.{hh, cc}` too. I'm not sure if using C++ macros there is a good idea. My guess is that it wouldn't look really good. For example, we would have to concatenate parts of the functions' names.